### PR TITLE
Rework connect spec versions and report metrics

### PR DESCRIPTION
**Pull request by dylan-apollo**
Reports some basic metrics on connectors usage and refactors connect spec version to an enum so we can make sure we're properly checking versions in the future (once we have more).

In theory also future-proofs some expansion code by including all relevant connect specs (not that we have more than one).

Draft until I can test these metrics.

[CNN-203]: https://apollographql.atlassian.net/browse/CNN-203?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by lennyburdette** [comment]
matt merged some test fixes if you want to rebase

File: apollo-federation/src/sources/connect/spec/mod.rs [hunk 1]
```diff
@@ -26,79 +26,88 @@ use apollo_compiler::Name;
...
-    pub(crate) fn new(version: Version, minimum_federation_version: Option<Version>) -> Self {
-        Self {
-            url: Url {
-                identity: Self::identity(),
-                version,
-            },
-            minimum_federation_version,
+    pub const fn version_str(&self) -> &'static str {
+        match self {
+            Self::V0_1 => "0.1",
```
Comment (lennyburdette) [inline apollo-federation/src/sources/connect/spec/mod.rs:52]: so the metric name would be `connectors.subgraphs_with_spec.0.1` — should we use a different delimiter like `0_1`?

File: apollo-router/src/plugins/connectors/tracing.rs [hunk 2]
```diff
@@ -1,2 +1,30 @@
...
+            let mut all_connectors = 0;
+            let mut unique_subgraphs = HashSet::default();
+            for connector in connectors {
+                all_connectors += 1;
+                unique_subgraphs.insert(connector.id.subgraph_name.clone());
+            }
+
+            u64_counter!(
+                "apollo.router.connectors",
+                "Number of connectors in the for a specific spec version",
```
Comment (lennyburdette) [inline apollo-router/src/plugins/connectors/tracing.rs:24]:
```suggestion
                "Number of connectors in supergraph the for a specific spec version",
```

**Comment by dylan-apollo** [comment]
Closing in favor of some new registry-based metrics

**Pull request closed without merging**